chia/wallet/util/compute_hints.py

```diff
@@ -1,22 +1,54 @@
 from __future__ import annotations
 
-from typing import List
+from dataclasses import dataclass
+from typing import Iterator
 
-from chia.types.blockchain_format.program import INFINITE_COST
+from clvm.casts import int_from_bytes
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.program import INFINITE_COST, Program
 from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.coin_spend import CoinSpend
 from chia.types.condition_opcodes import ConditionOpcode
+from chia.util.ints import uint64
+
+
+@dataclass(frozen=True)
+class HintedCoin:
+    coin: Coin
+    hint: bytes32
+
+
+def hinted_coin_from_condition(condition: Program, parent_coin_id: bytes32) -> HintedCoin:
+    data = condition.as_python()
+    if len(data) != 4 or data[0] != ConditionOpcode.CREATE_COIN:
+        raise ValueError(f"Not a hinted coin: {data}")
+
+    try:
+        coin = Coin(parent_coin_id, bytes32(data[1]), uint64(int_from_bytes(data[2])))
+    except Exception as e:
+        raise ValueError(f"Invalid coin: {e}, data: {data[1:2]}") from e
+
+    try:
+        hint = bytes32(data[3][0])
+    except Exception as e:
+        raise ValueError(f"Invalid hint: {e}, data: {data[3:]}") from e
+
+    return HintedCoin(coin, hint)
+
 
+def hinted_coins_in_coin_spend(coin_spend: CoinSpend) -> Iterator[HintedCoin]:
+    _, result_program = coin_spend.puzzle_reveal.run_with_cost(INFINITE_COST, coin_spend.solution)
+    parent_coin_id = coin_spend.coin.name()
+    for condition in result_program.as_iter():
+        try:
+            yield hinted_coin_from_condition(condition, parent_coin_id)
+        except Exception:
```

Comment:
It seems like you only throw value error from this function so maybe just catch that.  Also, perhaps it would be good to throw a specific exception because I imagine the caller would not like to accidentally consume an exception from an unexpectedly malformatted condition of some sort.

Reply:
I liked the behavior before because it's sort of a sanity check on the data we're processing.  This method is for processing coin spends, coins spends should only output valid conditions as part of their definition, so running this on something that doesn't should fail.  At the very least, as this PR sits, it would make the code less sane without adding anything to replace the functionality of checking that we're only processing valid coin spends.

Reply:
Okay agree, probably better to let other stuff still raise, added [25185d9](https://github.com/Chia-Network/chia-blockchain/pull/15274/commits/25185d93a7762ed7c6308379d6e1d24babf0038e).